documents/api/viewsets.py

```diff
@@ -316,18 +324,23 @@ def partial_update(self, request, pk, *args, **kwargs):
             attachment_serializer = CreateAttachmentSerializer(data=data)
             attachment_serializer.is_valid(raise_exception=True)
             attachment_serializer.save()
-        # Don't update history if patch has no data.
-        if request.data:
+        # Update history only if status changed.
+        if request.data.get("status"):
             status_history_serializer = StatusHistorySerializer(
                 data={
                     "document": document.id,
                     "value": document.status,
-                    "timestamp": document.updated_at,
+                    "timestamp": document.status_timestamp,
                 }
             )
             status_history_serializer.is_valid(raise_exception=True)
             status_history_serializer.save()
 
+            # Make sure the request data query dict is mutable, assign status_timestamp field to be updated.
+            request.data._mutable = True
```

Comment:
What are you doing here?

Reply:
In some situations request.data isn't mutable by default. The purpose here is to allow adding a new field to be saved with the serializer. If you have any alternative suggestions for this I would be glad to know.

Reply:
I would insert timezone.now() in Serializers update method straight to the right field